machine.go

```diff
@@ -675,19 +675,14 @@ func (h *Headscale) toNode(
 		[]netip.Prefix{},
 		addrs...) // we append the node own IP, as it is required by the clients
 
-	enabledRoutes, err := h.GetEnabledRoutes(&machine)
-	if err != nil {
-		return nil, err
-	}
-
-	allowedIPs = append(allowedIPs, enabledRoutes...)
-
 	primaryRoutes, err := h.getMachinePrimaryRoutes(&machine)
```

Comment:
> Also do not include secondary routes in allowedIPs, otherwise secondary routes might be used by nodes instead of the primary route.

I reckon that is not correct. Primary routes handled by that node are specified in the dedicated 
`PrimaryRoutes` field of `tailcfg.Node` https://github.com/juanfont/headscale/blob/main/machine.go#L741, even if we are including _all_ the enabled routes in `AllowedIPs`.

Reply:
This is what I assumed first after reading the structure's documentation. It states
```
// PrimaryRoutes are the routes from AllowedIPs that this node
// is currently the primary subnet router for, as determined
// by the control plane. It does not include the self address
// values from Addresses that are in AllowedIPs.
PrimaryRoutes []netip.Prefix `json:",omitempty"`
```

But it seems this field is used as an informative one so that subnet routers know that they are the primary router of an HA route. It is not used to populate the routing table of other peers... `AllowedIPs` is used for that.
I don't know Tailscale's internals as much as you, so I will let you double-check.